integration-tools/src/main/java/uk/ac/bbsrc/tgac/miso/integration/dp5mirage/DP5MirageScanner.java

```diff
@@ -0,0 +1,270 @@
+    this.getParamTimeout = getParamTimeout;
+  }
+
+  public void setPostActionTimeout(int postActionTimeout) {
+    this.postActionTimeout = postActionTimeout;
+  }
+
+  public void setDeleteActionTimeout(int deleteActionTimeout) {
+    this.deleteActionTimeout = deleteActionTimeout;
+  }
+
+  // Getters
+  public HttpClient getHttpClient() {
+    return this.httpClient;
+  }
+
+  private String getParameterAppend(Map<String,String> parameters) {
+    String appendQuery = "";
+
+    for (Entry<String, String> entry : parameters.entrySet()) {
+      appendQuery = appendQuery + entry.getKey() + "=" + entry.getValue() + "&";
+    }
+    return appendQuery;
+  }
+
+  /**
+   * Sending a GET request
+   * @param action
+   * @param urlStub
+   * @param parameters
+   * @return
+   * @throws URISyntaxException
+   */
+  private HttpRequest getGetParamRequest(String action, String urlStub, Map<String,String> parameters) throws URISyntaxException
+  {
+    URI uri = URI.create(urlStub + action);
+
+    if(parameters != null) {
+      String queryParam = getParameterAppend(parameters);
+      uri= new URI(uri.getScheme(), uri.getAuthority(),
+          uri.getPath(), queryParam, uri.getFragment());
+
+    }
+
+    request = HttpRequest.newBuilder().uri(uri)
+        .timeout(Duration.ofSeconds(getParamTimeout))
+        .GET().build();
+    return request;
+  }
+
+  /**
+   * Sending a POST request
+   * @param action
+   * @param urlStub
+   * @param parameters
+   * @return
+   * @throws URISyntaxException
+   */
+  private HttpRequest getPostParamRequest(String action, String urlStub, Map<String,String> parameters) throws URISyntaxException
+  {
+    URI uri = URI.create(urlStub + action);
+    if(parameters != null) {
+      String queryParam = getParameterAppend(parameters);
+      uri= new URI(uri.getScheme(), uri.getAuthority(),
+          uri.getPath(), queryParam, uri.getFragment());
+    }
+    request = HttpRequest.newBuilder().uri(uri)
+        .timeout(Duration.ofSeconds(postActionTimeout))
+        .POST(BodyPublishers.ofString("")).build();
+    return request;
+  }
+
+  /**
+   * Sending a PUT request
+   * @param action
+   * @param urlStub
+   * @return
+   * @throws URISyntaxException
+   */
+  private HttpRequest getPutParamRequest(String action, String urlStub) throws URISyntaxException
+  {
+    URI uri = URI.create(urlStub + action);
+    request = HttpRequest.newBuilder().uri(uri)
+        .timeout(Duration.ofSeconds(putActionTimeout))
+        .PUT(BodyPublishers.ofString("")).build();
+    return request;
+  }
+
+  /**
+   * Constructs a new DP5MirageScanner to communicate with a DP5-headless server and retrieve scan
+   * data
+   * <p>
+   * Initialise the host and port which is localhost and 8777 by default
+   * @throws IntegrationException if the server hostname or ip could not be resolved
+   */
+  public DP5MirageScanner() throws IntegrationException {
+    this.host = "localhost";
+    this.port = 8777;
```

Comment:
Having a scanner and MISO running on the same machine would be weird. I don't think there should be a default constructor since there are no defaults that make sense